### PR TITLE
[lc/dif] Fix `claim_transition_if` claim value

### PR DESCRIPTION
**Pull request by cindychip**
This PR fixes the value to claim the mutex. From the design spec, we are
using `A5` to claim instead of `1`.

Signed-off-by: Cindy Chen <chencindy@opentitan.org>

### CODE REVIEW

**Review by msfschaffner — APPROVED** [review]

**Review by moidx — CHANGES_REQUESTED** [review]
Thanks @cindychip. You will also have to update the unit test with the expected values. 

https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/dif/dif_lc_ctrl_unittest.cc;l=169

File: sw/device/lib/dif/dif_lc_ctrl_unittest.cc [hunk 1]
```diff
@@ -166,17 +166,17 @@ TEST_F(AlertTest, NullArgs) {
...
 TEST_F(MutexTest, Acquire) {
-  EXPECT_WRITE32(LC_CTRL_CLAIM_TRANSITION_IF_REG_OFFSET, true);
-  EXPECT_READ32(LC_CTRL_CLAIM_TRANSITION_IF_REG_OFFSET, true);
+  EXPECT_WRITE32(LC_CTRL_CLAIM_TRANSITION_IF_REG_OFFSET, 0xa5);
+  EXPECT_READ32(LC_CTRL_CLAIM_TRANSITION_IF_REG_OFFSET, 0xa5);
   EXPECT_EQ(dif_lc_ctrl_mutex_try_acquire(&lc_), kDifLcCtrlMutexOk);
 
-  EXPECT_WRITE32(LC_CTRL_CLAIM_TRANSITION_IF_REG_OFFSET, true);
+  EXPECT_WRITE32(LC_CTRL_CLAIM_TRANSITION_IF_REG_OFFSET, 0xa5);
   EXPECT_READ32(LC_CTRL_CLAIM_TRANSITION_IF_REG_OFFSET, false);
```
Comment (moidx) [inline sw/device/lib/dif/dif_lc_ctrl_unittest.cc:174]: You can also update these to 0 to keep things consistent with the true encoding (0xa5). btw, is 0 -> 0x5a or some other value?

**Review by moidx — APPROVED** [review]

File: sw/device/lib/dif/dif_lc_ctrl_unittest.cc (hunk 1 above)
Comment (cindychip) [reply sw/device/lib/dif/dif_lc_ctrl_unittest.cc:174]: I see, I will update to 0. Thanks for the suggestion. Yes from the documentation, it is 0 -> 0x5a. https://docs.opentitan.org/hw/ip/lc_ctrl/doc/index.html#Reg_claim_transition_if

File: sw/device/lib/dif/dif_lc_ctrl.c [hunk 2]
```diff
@@ -238,7 +238,7 @@ dif_lc_ctrl_mutex_result_t dif_lc_ctrl_mutex_try_acquire(
   }
 
   mmio_region_write32(lc->params.base_addr,
-                      LC_CTRL_CLAIM_TRANSITION_IF_REG_OFFSET, 1);
+                      LC_CTRL_CLAIM_TRANSITION_IF_REG_OFFSET, 0xa5);
```
Comment (moidx) [inline sw/device/lib/dif/dif_lc_ctrl.c:241]: I think `dif_lc_ctrl_mutex_release()` will also have to be updated then. 

I recommend to encode these variables then at the top of this file: 

```cc
enum {
  /** foobar and maybe reference to spec */
  kLcCtrlMutexAcquire = 0xa5,
  /** foobar */
  kLcCtrlMutexRelease = 0x5a,
};
```

Comment (cindychip) [reply sw/device/lib/dif/dif_lc_ctrl.c:241]: I think the dif_lc_ctrl_mutex_release() is correct. But down for adding an enum for that. From this spec [here](https://docs.opentitan.org/hw/ip/lc_ctrl/doc/index.html#Reg_claim_transition_if), I believe the enum could be declared as:
```
enum {
  /** foobar and maybe reference to spec */
  kLcCtrlMutexAcquire = 0xa5,
  /** foobar */
  kLcCtrlMutexRelease = 0,
};
```

Comment (moidx) [reply sw/device/lib/dif/dif_lc_ctrl.c:241]: sounds good. This will help to cross-link to the spec. thanks!

Comment (cindychip) [reply sw/device/lib/dif/dif_lc_ctrl.c]: Oops, i did not see that, let me add the link to the spec.

File: sw/device/lib/dif/dif_lc_ctrl.h [hunk 3]
```diff
@@ -208,6 +208,22 @@ typedef enum dif_lc_ctrl_id_state {
   kDifLcCtrlIdStateInvalid,
 } dif_lc_ctrl_id_state_t;
 
+/**
+ * A lifecycle controller mutex acquire and release code.
+ */
+typedef enum dif_lc_ctrl_mutex_code {
+  /**
+   * According to LC_CTRL register spec [hw/ip/lc_ctrl/data/lc_ctrl.hjson],
```
Comment (cindychip) [inline sw/device/lib/dif/dif_lc_ctrl.h:216]: Hi Miguel @moidx  sorry I did not find any examples for referring to HW spec. Could you confirm if this is the correct syntax for SW?

Comment (moidx) [reply sw/device/lib/dif/dif_lc_ctrl.h:216]: You can add a link directly to the generated docs like it is done here: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/dif/dif_csrng.c;l=21

File: sw/device/lib/dif/dif_lc_ctrl.h [hunk 4]
```diff
@@ -208,6 +208,22 @@ typedef enum dif_lc_ctrl_id_state {
   kDifLcCtrlIdStateInvalid,
 } dif_lc_ctrl_id_state_t;
 
+/**
+ * A lifecycle controller mutex acquire and release code.
+ */
+typedef enum dif_lc_ctrl_mutex_code {
```
Comment (moidx) [inline sw/device/lib/dif/dif_lc_ctrl.h:214]: One last nit: You can move this to `dif_lc_ctrl.c` as this type is not exposed on an external interface. You don't need to associate a type with it: 

```c
enum {
  kLcCtrlMutexAcquire = 0xa5,
  kLcCtrlMutexRelease = 0,
};
```

In the unit test it is ok to hardcode the expected values. It is usually preferred to have this level of duplication when writing unittests.

Comment (cindychip) [reply sw/device/lib/dif/dif_lc_ctrl.h:214]: Oh I see, thanks for pointing that out. I will update the code.

File: sw/device/lib/dif/dif_lc_ctrl.h (hunk 3 above)
Comment (cindychip) [reply sw/device/lib/dif/dif_lc_ctrl.h:216]: Thanks for the reference. I will follow that.

**Pull request merged**